pkg/cert/generatecerts.go

```diff
@@ -0,0 +1,225 @@
+	if err != nil {
+		logrus.Printf("failed to generate RSA key for intermediate certificate. Error - %vs", err.Error())
+		return err
+	}
+
+	interCertTemplate := x509.Certificate{
+		Subject: pkix.Name{
+			Organization: []string{"Intelops"},
+			CommonName:   "Optimizor CA",
+			Locality:     []string{"agent"},
+		},
+		SerialNumber:          big.NewInt(1),
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().AddDate(1, 0, 0),
+		IsCA:                  true,
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		BasicConstraintsValid: true,
+	}
+
+	interCert, err := x509.CreateCertificate(rand.Reader, &interCertTemplate, &rootCertTemplate, &interKey.PublicKey, rootKey)
+	if err != nil {
+		logrus.Printf("failed to create intermediate certificate. Error - %vs", err.Error())
+		return err
+	}
+
+	interCertPEM := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: interCert})
+	err = ioutil.WriteFile("certs/intermediate/ca-cert.pem", interCertPEM, 0644)
+	if err != nil {
+		logrus.Printf("error while writing from intermediate cert to certs/intermediate/ca-cert.pem. Error - %vs", err.Error())
+		return err
+	}
+
+	interKeyPEM := pem.EncodeToMemory(&pem.Block{Type: "RSA PRIVATE KEY", Bytes: x509.MarshalPKCS1PrivateKey(interKey)})
+	err = ioutil.WriteFile("certs/intermediate/ca-key.pem", interKeyPEM, 0644)
+	if err != nil {
+		logrus.Printf("error while writing from intermediate key to certs/intermediate/ca-key.pem. Error - %vs", err.Error())
+		return err
+	}
+
+	catCertPEM, err := ioutil.ReadFile("certs/root/root-cert.pem")
+	if err != nil {
+		logrus.Printf("error while reading certs/root/root-cert.pem file . Error - %vs", err.Error())
+		return err
+	}
+
+	err = ioutil.WriteFile("certs/intermediate/cert-chain.pem", append(interCertPEM, catCertPEM...), 0644)
+	if err != nil {
+		logrus.Printf("error while writing to certs/intermediate/cert-chain.pem file. Error - %vs", err.Error())
+		return err
+	}
+
+	err = ioutil.WriteFile("certs/intermediate/root-cert.pem", catCertPEM, 0644)
+	if err != nil {
+		logrus.Printf("error while writing to certs/intermediate/root-cert.pem file. Error - %vs", err.Error())
+		return err
+	}
+
+	// Server certificate creation
+	serverKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		logrus.Printf("failed to generate RSA key for server certificate. Error - %vs", err.Error())
+		return err
+	}
+
+	serverCertTemplate := x509.Certificate{
+		Subject: pkix.Name{
+			Organization: []string{"Intelops Inc."},
+			CommonName:   "*.dev.optimizor.app",
+		},
+		SerialNumber:          big.NewInt(1),
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().AddDate(0, 1, 0),
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageKeyEncipherment,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		DNSNames:              []string{"captenagent.dev.optimizor.app", "dev.optimizor.app", "localhost"},
```

Comment:
don't hard code.

use domain-name from capten.yaml, and fixed names for agent.


make agent host name as "agent" with const string
agent.{domain name}